iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableLambdaEndpoint.java

```diff
@@ -19,7 +20,21 @@
     @Override
     public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
         Message request = serializer.deserialize(input, Message.class);
-        Message response = processor.processMessage(request);
-        serializer.serialize(output, response);
+
+        try {
+            Message message = processor.processMessage(request);
+            SuccessResponse success = new SuccessResponse(200, message);
+            serializer.serialize(output, success);
+        } catch (TooManyRequestsException e) {
+            Map<String, String> body = new HashMap<>();
+            body.put("message", "Iterable rate limit exceeded");
+            ErrorResponse error = new ErrorResponse(429, body);
+            serializer.serialize(output, error);
+        } catch (IOException e) {
+            Map<String, String> body = new HashMap<>();
+            body.put("message", e.getMessage());
+            ErrorResponse error = new ErrorResponse(500, body);
```

Comment:
I think if we return this, Lambda metrics will start showing those requests as successful because they don't throw exceptions. It may be better to let the exception through for monitoring unless we know we can track them downstream.

Reply:
That's a good call. I'll stop catching those exceptions unless we get another form of monitoring setup.